pkg/controller/pxc/upgrade.go

```diff
@@ -127,14 +127,27 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 	var newContainers []corev1.Container
 	var newInitContainers []corev1.Container
 
-	secrets := cr.Spec.SecretsName
+	secretsName := cr.Spec.SecretsName
 	if cr.CompareVersionWith("1.6.0") >= 0 {
-		secrets = "internal-" + cr.Name
+		secretsName = "internal-" + cr.Name
 	}
 
 	// pmm container
 	if cr.Spec.PMM != nil && cr.Spec.PMM.Enabled {
-		pmmC, err := sfs.PMMContainer(cr.Spec.PMM, secrets, cr)
+		secret := new(corev1.Secret)
+		err := r.client.Get(context.TODO(), types.NamespacedName{
+			Name: secretsName, Namespace: cr.Namespace,
+		}, secret)
+		if client.IgnoreNotFound(err) != nil {
+			return errors.Wrap(err, "get internal secret")
+		}
+		useAPI := true
+		if _, ok := secret.Data["pmmserverkey"]; !ok {
+			if _, ok := secret.Data["pmmserver"]; ok {
+				useAPI = false
+			}
+		}
```

Comment:
why we can't move this logic to PMMContainer

Reply:
fixed https://github.com/percona/percona-xtradb-cluster-operator/pull/1119/commits/ec722f874a7389f98f366a46adeee384fccd9775